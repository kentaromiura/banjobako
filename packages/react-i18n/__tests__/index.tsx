import React, { FC, useContext } from 'react';
import renderer from 'react-test-renderer';
import I18n, { Context, Provider, init, I18nRenderJS } from '../index';

const TestComponent: FC<{}> = () => {
  const i18n = useContext(Context);
  return <>{i18n.t('hello')}</>;
};

test('Render I18n with using locales', () => {
  const headers = {
    'accept-language': 'en'
  };
  const locales = {
    en: {
      hello: 'Hello'
    },
    ja: {
      hello: 'こんにちは'
    }
  };
  window.__SIMPLE_I18N__ = { resource: locales.en, lang: 'en' };

  const i18n = init({
    headers,
    locales
  });
  expect(
    renderer
      .create(
        <Provider value={i18n}>
          <I18n id="hello" />
        </Provider>
      )
      .toJSON()
  ).toMatchSnapshot();

  expect(
    renderer
      .create(
        <Provider value={i18n}>
          <TestComponent />
        </Provider>
      )
      .toJSON()
  ).toMatchSnapshot();

  expect(
    renderer
      .create(<I18nRenderJS headers={headers} locales={locales} />)
      .toJSON()
  ).toMatchSnapshot();
});
