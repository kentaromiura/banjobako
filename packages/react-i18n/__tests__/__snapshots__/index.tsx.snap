// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render I18n with using locales 1`] = `"Hello"`;

exports[`Render I18n with using locales 2`] = `"Hello"`;

exports[`Render I18n with using locales 3`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "window.__SIMPLE_I18N__={resource: {\\"hello\\":\\"Hello\\"}, lang: \\"en\\"}",
    }
  }
/>
`;
